atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/PostgresConnectionConfig.java

```diff
@@ -105,4 +105,9 @@ public final DBType getDbType() {
     public final String type() {
         return TYPE;
     }
+
+    @Override
+    public Optional<String> getConnectionInitSql() {
+        return Optional.of("SET statement_timeout = " + (getSocketTimeoutSeconds() * 1000));
```

Comment:
instead of reusing the socket timeout can we add a new statementTimeout param? The default value of the param can use the socketTimeout

Reply:
(Separately, do we want to expose this to Atlas clients based on things they provide in the config? Maybe we should have the timeout as a param and set this inside the ConnectionConfig)

Reply:
Not totally clear on what you mean here - we'll allow statement-timeout to be specified in config, and we'll set the init-string based connection (for postgres) based on that. Clients could override the init-string as well, but if they do they'll need to consider the statement-timeout.